### PR TITLE
Rtc/am1805

### PR DESCRIPTION
**Pull request by srisurya1**

### CODE REVIEW

File: dts/bindings/rtc/artasie,am1805.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+include:
+  - name: rtc-device.yaml
+  - name: i2c-device.yaml
+  - name: pm.yaml
+
+properties:
+  reg:
+    required: true
+
+  am1805-gpios:
```
Comment (parthitce) [inline dts/bindings/rtc/artasie,am1805.yaml:18]: call it as `int-gpios`

File: dts/bindings/rtc/artasie,am1805.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# PDX-License-Identifier: Apache-2.0
+
+description: ARTASIE AM1805 RTC
+
+compatible: "artasie,am1805"
+
+include:
+  - name: rtc-device.yaml
+  - name: i2c-device.yaml
+  - name: pm.yaml
```
Comment (parthitce) [inline dts/bindings/rtc/artasie,am1805.yaml:12]: I don't see power managements ops in the driver. Do we need it?

File: dts/bindings/rtc/artasie,am1805.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2023 Linumiz
+# Author : Sri Surya <srisurya@linumiz.com>
+# PDX-License-Identifier: Apache-2.0
```
Comment (parthitce) [inline dts/bindings/rtc/artasie,am1805.yaml:3]: typo and move above

File: dts/bindings/rtc/artasie,am1805.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+compatible: "artasie,am1805"
+
+include:
+  - name: rtc-device.yaml
+  - name: i2c-device.yaml
+  - name: pm.yaml
+
+properties:
+  reg:
```
Comment (parthitce) [inline dts/bindings/rtc/artasie,am1805.yaml:15]: IMO it's part of i2c-device.yaml and we don't need that in the consumer. Not sure!

File: drivers/rtc/rtc_am1805.c [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+
+struct am1805_data {
+	struct k_mutex lock;
+#if defined(CONFIG_RTC_ALARM) && defined(AM1805_GPIO_IN_USE)
+	rtc_alarm_callback alarm_user_callback;
+	void *alarm_user_data;
+	/*For gpio-interrupt*/
+	struct gpio_callback am1805_callback;
+	struct k_thread am1805_thread;
+	struct k_sem am1805_sem;
```
Comment (parthitce) [inline drivers/rtc/rtc_am1805.c:105]: `int_sem` ?

**Review by parthitce — COMMENTED** [review]
IMO it makes sense to mention the backward compat with another IC which uses the same IP and also point the datasheet in the commit message.

File: dts/bindings/rtc/artasie,am1805.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2023 Linumiz
+# Author : Sri Surya <srisurya@linumiz.com>
+# PDX-License-Identifier: Apache-2.0
+
+description: ARTASIE AM1805 RTC
+
+compatible: "artasie,am1805"
```
Comment (parthitce) [inline dts/bindings/rtc/artasie,am1805.yaml:7]: Also needs to be added in `dts/bindings/vendor-prefixes.txt`

File: dts/bindings/vendor-prefixes.txt [hunk 7]
```diff
@@ -66,6 +66,7 @@ aries	Aries Embedded GmbH
 arm	ARM Ltd.
 armadeus	ARMadeus Systems SARL
 arrow	Arrow Electronics
+ambiq	Ambiq Micro
```
Comment (parthitce) [inline dts/bindings/vendor-prefixes.txt:69]: Alpha order

**Comment by parthitce** [comment]
Otherwise, looks good for me

**Comment by ssekar15** [comment]
looks fine for me.. please create upstream PR

**Pull request closed without merging**